### ISSUE TITLE
keras.ops.categorical_crossentropy ignores axis argument in symbolic compute_output_spec

### ISSUE THREAD

**Issue by buildwithsuhana** [keras-team/keras#23740]
When calling `keras.ops.categorical_crossentropy` on symbolic `KerasTensor` inputs with a non-default `axis` (e.g., `axis=1` or `axis=-2`), `CategoricalCrossentropy.compute_output_spec` ignores `self.axis` and always strips the last dimension (`output.shape[:-1]`).
As a result, symbolic execution returns the wrong output shape whenever `axis != -1` and fails to raise a `ValueError` when `axis` is out of bounds.
### Root Cause
In `CategoricalCrossentropy.compute_output_spec` in [`keras/src/ops/nn.py`](https://github.com/keras-team/keras/blob/master/keras/src/ops/nn.py) (line 2085):
```python
def compute_output_spec(self, target, output):
    ...
    return KerasTensor(output.shape[:-1], dtype=output.dtype)
```

Unlike SparseCategoricalCrossentropy.compute_output_spec immediately below it (which canonicalizes self.axis via canonicalize_axis(self.axis, len(output.shape)) and returns output.shape[:axis] + output.shape[axis + 1 :]), CategoricalCrossentropy.compute_output_spec hardcodes output.shape[:-1].

Reproduction colab: https://colab.research.google.com/drive/1ieqLsG4DEnXpgTyLxQTA_k-4hERBdXBC?usp=sharing
